coursier.bzl

```diff
@@ -260,14 +274,14 @@ def _coursier_fetch_impl(repository_ctx):
     if repository_ctx.attr.fetch_sources:
         cmd = _generate_coursier_command(repository_ctx)
         cmd.extend(["fetch"])
-        cmd.extend(repository_ctx.attr.artifacts)
+        cmd.extend(artifact_coordinates)
         cmd.extend(["--artifact-type", "jar,aar,bundle,src"])
         cmd.append("--quiet")
         cmd.append("--no-default")
         cmd.extend(["--sources", "true"])
         cmd.extend(["--json-output-file", "src-dep-tree.json"])
-        for repository in repository_ctx.attr.repositories:
-            cmd.extend(["--repository", repository])
+        for repository in repositories:
+            cmd.extend(["--repository", repository["repo_url"]])
```

Comment:
Please add support for http basic authentication. Since the urls were passed as-is before this, you could use "http://user:password@example.com". That information is lost here now.

Reply:
Same for above.

Reply:
Yep, meant to do that and forgot.

Reply:
thanks for following up! it looks like it still uses the repo_url value without credentials?

Reply:
Ohhhhh, you wanted me to actually *use* the function I wrote. 
🤦‍♂️

Reply:
😄 thanks!